netty/src/main/java/io/grpc/netty/ProtocolNegotiator.java

```diff
@@ -56,4 +58,9 @@
    * has completed successfully, the provided handler is installed.
    */
   Handler newHandler(GrpcHttp2ConnectionHandler handler);
+
+  /**
+   * Returns a set of attributes, which varies depending on the negotiator implementation.
+   */
+  Attributes getAttributes();
```

Comment:
On server-side, the negotiator calls `grpcHandler.handleProtocolNegotiationCompleted` on the handler passed to `newHandler()`. That's important, as each connection will have a different `Handler` and can have different `Attributes`. I don't think we could actually implement a useful implementation of this `ProtocolNegotiator.getAttributes()` because it isn't bound to a connection (the `Handler` is, but not the negotiator).